tableaudocumentapi/datasource.py

```diff
@@ -235,6 +235,56 @@ def clear_repository_location(self):
             self._datasourceXML.remove(tag)
 
     ###########
+    # Folders
+    ###########
+
+    @property
+    def folders(self):
+        """ Returns all folders.
+        """
+        return self._datasourceTree.findall('.//folder')
+
+    def add_folder(self, name, role, fields):
+        """ Adds a folder field with the given fields.
+
+        Args:
+            name:  Name of the new folder. String.
+            role:  Role of the new folder. String.
+            fields:  Fields of the new folder. String.
+
+        Returns:
+            The new calculated folder that was created. ET.Element.
+        """
+        # TODO: It might be better to create a dedicated "Folder" object:
```

Comment:
I know you're working on a Folder class now -- what we didn't discuss is how we initialize all existing folders and the fields therein for those objects, a good model might be what we do with Datasources in workbooks, or Worksheets